c++/src/kj/CMakeLists.txt

```diff
@@ -200,8 +221,9 @@ if(BUILD_TESTING)
       parse/char-test.c++
       compat/url-test.c++
       compat/http-test.c++
+      compat/gzip-test.c++
     )
-    target_link_libraries(kj-heavy-tests kj-http kj-async kj-test kj)
+    target_link_libraries(kj-heavy-tests kj-http kj-gzip z kj-async kj-test kj)
```

Comment:
I think you need to make kj-gzip and the inclusion of compat/gzip-test.c++ conditional on ZLIB_FOUND. This appears to be why the CI build fails on Windows.

Reply:
Maybe... Although that's odd. Both gzip.c++ and gzip-test.c++ are wrapped into `#if KJ_HAS_ZLIB` so it shouldn't even try to include zlib symbols when that variable is not defined, and I'm defining it conditionally on `ZLIB_FOUND` already.

Reply:
Ah, the reason is rather that `add_library` for `kj_gzip` is conditional, but `target_link_libraries` is not, so it can't find the original definition of `kj-gzip` library, and complains about that and not about zlib.

Reply:
Tried to fix it by declaring `add_library` unconditionally, but adding define symbol conditionally, so that linking to kj-gzip itself always works, but when zlib is not found, it's no-op. Still, Windows build is failing, and in fact it explicitly says that it can't find `z.lib` as if `ZLIB_FOUND` was successful but zlib actually doesn't exist or something... https://ci.appveyor.com/project/kentonv/capnproto/build/650/job/wshgjjno03juk5s2

Reply:
Oh nevermind, found it, it's the explicit `z` leftover in list of libraries.

Reply:
Seems fixed now, yay. But please don't merge yet, I'd like to squash these CMake fixes together.

Reply:
Rebased.